spec/logical_param_match_spec.rb

```diff
@@ -0,0 +1,218 @@
+require_relative 'shared'
+
+describe 'logical_param_matching1.sag' do
+  include_context 'component test', 'fixtures/logical_param_matching1.sag'
+
+  include_examples 'lexing', [
+    Token.new(:identifier, 'main'),
+    Token.new(:return),
+    Token.new(:type, :INT),
+    Token.new(:entry_function_line),
+    Token.new(:identifier, 'main'),
+    Token.new(:return),
+    Token.new(:function_call, 'round'),
+    Token.new(:integer_constant, 16),
+    Token.new(:end),
+
+    Token.new(:identifier, 'round'),
+    Token.new(:type, :INT),
+    Token.new(:return),
+    Token.new(:type, :INT),
+    Token.new(:function_line),
+    Token.new(:identifier, 'round'),
+    Token.new(:variable, 'X'),
+    Token.new(:condition),
+    Token.new(:open_expression),
+    Token.new(:variable, 'X'),
+    Token.new(:function_call, '%'),
+    Token.new(:integer_constant, 10),
+    Token.new(:close_expression),
+    Token.new(:function_call, '<'),
+    Token.new(:integer_constant, 5),
+    Token.new(:return),
+    Token.new(:variable, 'X'),
+    Token.new(:function_call, '-'),
+    Token.new(:open_expression),
+    Token.new(:variable, 'X'),
+    Token.new(:function_call, '%'),
+    Token.new(:integer_constant, 10),
+    Token.new(:close_expression),
+    Token.new(:break),
+    Token.new(:identifier, 'round'),
+    Token.new(:variable, 'X'),
+    Token.new(:return),
+    Token.new(:variable, 'X'),
+    Token.new(:function_call, '+'),
+    Token.new(:open_expression),
+    Token.new(:integer_constant, 10),
+    Token.new(:function_call, '-'),
+    Token.new(:open_expression),
+    Token.new(:variable, 'X'),
+    Token.new(:function_call, '%'),
+    Token.new(:integer_constant, 10),
+    Token.new(:close_expression),
+    Token.new(:end)
+  ]
```

Comment:
maybe an 'contains' shared example would be easier in come cases where we're more interested in new tokens getting parsed. Whether it checks for a just one token or a subset, could be decided later